Analysis/Tasks/PWGHF/HFCandidateCreatorBPlus.cxx

```diff
@@ -0,0 +1,286 @@
+        const auto& BSecVertex = bfitter.getPCACandidate();
+        auto chi2PCA = bfitter.getChi2AtPCACandidate();
+        auto covMatrixPCA = bfitter.calcPCACovMatrix().Array();
+        hCovSVXX->Fill(covMatrixPCA[0]); //FIXME: Calculation of errorDecayLength(XY) gives wrong values without this line.
+
+        pVecBCand = RecoDecay::PVec(pVecD0, pVecBach);
+
+        // get track impact parameters
+        // This modifies track momenta!
+        auto primaryVertex = getPrimaryVertex(collision);
+        auto covMatrixPV = primaryVertex.getCov();
+        hCovPVXX->Fill(covMatrixPV[0]);
+        o2::dataformats::DCA impactParameter0;
+        o2::dataformats::DCA impactParameter1;
+
+        bfitter.getTrack(0).propagateToDCA(primaryVertex, d_bz, &impactParameter0);
+        bfitter.getTrack(1).propagateToDCA(primaryVertex, d_bz, &impactParameter1);
+
+        // get uncertainty of the decay length
+        double phi, theta;
+        getPointDirection(array{collision.posX(), collision.posY(), collision.posZ()}, BSecVertex, phi, theta);
+        auto errorDecayLength = std::sqrt(getRotatedCovMatrixXX(covMatrixPV, phi, theta) + getRotatedCovMatrixXX(covMatrixPCA, phi, theta));
+        auto errorDecayLengthXY = std::sqrt(getRotatedCovMatrixXX(covMatrixPV, phi, 0.) + getRotatedCovMatrixXX(covMatrixPCA, phi, 0.));
+
+        int hfFlag = 1 << hf_cand_bplus::DecayType::BPlusToD0Pi;
+
+        // fill candidate table rows
+        rowCandidateBase(collision.globalIndex(),
+                         collision.posX(), collision.posY(), collision.posZ(),
+                         BSecVertex[0], BSecVertex[1], BSecVertex[2],
+                         errorDecayLength, errorDecayLengthXY,
+                         chi2PCA,
+                         pVecD0[0], pVecD0[1], pVecD0[2],
+                         pVecBach[0], pVecBach[1], pVecBach[2],
+                         impactParameter0.getY(), impactParameter1.getY(),
+                         std::sqrt(impactParameter0.getSigmaY2()), std::sqrt(impactParameter1.getSigmaY2()),
+                         candidate.globalIndex(), track.globalIndex(), //index D0 and bachelor
+                         hfFlag);
+      } //track loop
+    }   //D0 cand loop
+  }     //process
+};      //struct
+
+/// Extends the base table with expression columns.
+struct HfCandidateCreatorBplusExpressions {
+  Spawns<aod::HfCandBPlusExt> rowCandidateBPlus;
+  void init(InitContext const&) {}
+};
+
+/// Performs MC matching.
+struct HfCandidateCreatorBplusMc {
+  Produces<aod::HfCandBPMCRec> rowMCMatchRec;
+  Produces<aod::HfCandBPMCGen> rowMCMatchGen;
+
+  void process(aod::HfCandBPlus const& candidates,
+               aod::HfCandProng2,
+               aod::BigTracksMC const& tracks,
+               aod::McParticles const& particlesMC)
+  {
+    int indexRec = -1, indexRecD0 = -1;
+    int8_t signB = 0, signD0 = 0;
+    int8_t flag = 0;
+    int kD0pdg = pdg::Code::kD0;
```

Comment:
`kD0pdg` is not used consistently.

Reply:
Do you mean I create a new Int_t kD0pdg instead of using pdg::Code::kD0 directly? I tried it but the compilation failed. It worked only when I pass it as an int instead of enum.